test/e2e/preset_test.go

```diff
@@ -86,6 +89,14 @@ func createCustomWorkspaceWithAdapter(numOfNode int) *kaitov1alpha1.Workspace {
 	return workspaceObj
 }
 
+func updateCustomWorkspaceWithAdapter(workspaceObj *kaitov1alpha1.Workspace, validAdapters []kaitov1alpha1.AdapterSpec) *kaitov1alpha1.Workspace {
+	By("Updating a workspace with adapter", func() {
+		workspaceObj.Inference.Adapters = validAdapters
+		updateAndValidateWorkspace(workspaceObj)
```

Comment:
updateAndValidateWorkspace is just called once so I think the entire updateAndValidateWorkspace function can be "inline" here.